_posts/2021-07-22-using-kubernetes-config-provider-to-load-data-from-secrets-and-config-maps.md

```diff
@@ -0,0 +1,333 @@
+---
+layout: post
+title:  "Using Kubernetes Configuration Provider to load data from Secrets and Config Maps"
+date: 2021-07-22
+author: jakub_scholz
+---
+
+When running Apache Kafka on Kubernetes, you will sooner or later probably need to use Config Maps or Secrets.
+Either to store something in them, or load them into your Kafka configuration.
+That is true regardless of whether you use Strimzi to manage your Apache Kafka cluster or something else.
+Kubernetes has its own way of using Secrets and Config Maps from Pods.
+But they might not be always sufficient.
+That is why in Strimzi, we created Kubernetes Configuration Provider for Apache Kafka which we will introduce in this blog post.
+
+<!--more-->
+
+Usually, when you need to use data from a Config Map or Secret in your Pod, you will either mount it as volume or map it to an environment variable.
+Both methods are configured in the `spec` section or the Pod resource or in the `spec.template.spec` section when using higher level resources such as Deployments or StatefulSets.
+
+When mounted as a volume, the contents of the Secret or Config Map will appear as files inside your Pod and its containers.
+Your application can just read and work with them as with any other files.
+The path where the files appear is configurable in the Pod specification, so you can mount them exactly where your application would expect them.
+You can also mount only selected files / fields instead of mounting the whole Secret or Config Map.
+When the contents of the Secret or Config Map changes, the files inside the Pod will be automatically updated (this does not happen immediately but in periodical intervals).
```

Comment:
When content of Secret or Config Map changes, files inside the Pod will also be automatically updated. 
(feel free to ignore my suggestions, as I am not native speaker :) )